x-pack/plugins/search_indices/public/components/start/create_index_code.tsx

```diff
@@ -47,18 +48,48 @@ export const CreateIndexCodeView = ({
     [usageTracker, changeCodingLanguage]
   );
   const elasticsearchUrl = useElasticsearchUrl();
+  const { apiKey, apiKeyIsVisible } = useSearchApiKey();
+
   const codeParams = useMemo(() => {
     return {
       indexName: createIndexForm.indexName || undefined,
       elasticsearchURL: elasticsearchUrl,
+      apiKey: apiKeyIsVisible && apiKey ? apiKey : undefined,
```

Comment:
we'll need to do this for the index details no data view too [here](https://github.com/elastic/kibana/blob/main/x-pack/plugins/search_indices/public/components/index_documents/add_documents_code_example.tsx#L61-L69)

Reply:
yep was done as part of the change https://github.com/elastic/kibana/blob/cd87f441ef7c4dccb448fa6b48fc102e7b67e465/x-pack/plugins/search_indices/public/components/index_documents/add_documents_code_example.tsx#L70